tests/lib/rules/no-extra-parens.js

```diff
@@ -466,7 +466,9 @@ ruleTester.run("no-extra-parens", rule, {
         "for ((let) in foo);",
         "for ((let[foo]) in bar);",
         "for ((let)[foo] in bar);",
-        "for ((let[foo].bar) in baz);"
+        "for ((let[foo].bar) in baz);",
+
+        "let s = `${(a, b)}`"
```

Comment:
This set of parentheses seems unnecessary. Am I missing something?

Reply:
Good catch.

Reply:
However, `SequenceExpression` has been checked: https://github.com/eslint/eslint/blob/e5f1ccc9e2d07ad0acf149027ffc382021d54da1/lib/rules/no-extra-parens.js#L683 .

Reply:
Fixed.

Reply:
Actually, maybe these parentheses should stay, just to avoid conflicts with the `no-sequences` rule which allows only explicitly parenthesized sequences (and some exceptions, like for-loop).

See [no-sequences docs](https://eslint.org/docs/rules/no-sequences) and #561 , the idea is that these parentheses might mean _intent_. So perhaps this rule should never warn/remove parentheses around a `SequenceExpression`? These parens will usually stay because of its lowest precedence, but there are already examples where the parens would be removed.

With all that said, I believe the last commit is correct, this logic should be in one place, like `ruleApplies()`, not in the handlers.

If this makes any sense I could open an issue?

Reply:
That's a tough one. I wonder if this rule needs to have an option to support removing parentheses around sequence expressions (because not everyone would use no-sequences and expect the parentheses behavior).

I'm okay with having this fix not worry about sequence expressions for now and discussing in a separate issue. I think there are a few other useful changes in this PR already.

@g-plane @mdjermanovic What do you think?

Reply:
Completely agree. My post was confusing, I believe that in [47829be](https://github.com/eslint/eslint/pull/11909/commits/47829be39d1515838fa4c2143eb875b4915bde5f) both rule and the test case are correctly representing the current rule's logic regarding this matter.

Reply:
Whether adding an option or not is OK but this should be discussed at another new issue. Let's focus on: should SequenceExpression be reported and be fixed?

Reply:
I think we should preserve the old behavior for now. I think that means don't report (or fix) SequenceExpression nodes.

Reply:
I apologize for the confusion, my post turned out to be completely irrelevant for this PR.

SequenceExpression nodes are already being reported and fixed wherever the removal does not break syntax or semantics, i.e. wherever the parens are not necessary, for example:

* `(a, b);`
* `if ((a, b));`
* `while ((a, b));`
* `for ((a, b); (a, b); (a, b));` - all three

So I guess that this should be reported and fixed too:

```js
`${(a, b)}`
```

Reply:
So keep the latest commit?

Reply:
> So keep the latest commit?

I would agree with that.